functions/zoom-uploader.py

```diff
@@ -124,27 +130,58 @@ def s3_prefix(self):
     def zoom_series_id(self):
         return self.data['meeting_number']
 
+    def series_id_from_schedule(self):
+
+        dynamodb = boto3.resource('dynamodb')
+        table = dynamodb.Table(CLASS_SCHEDULE_TABLE)
+
+        r = table.get_item(
+            Key={"zoom_series_id": str(self.zoom_series_id)}
+        )
+
+        if 'Item' not in r:
+            return None
+        else:
+            schedule = r['Item']
+            logger.info(schedule)
+
+        zoom_time = datetime.strptime(self.created_local, '%Y-%m-%dT%H:%M:%SZ')
+        weekdays = ['M', 'T', 'W', 'R', 'F']
+        if zoom_time.weekday() > 4:
+            logger.debug("Meeting occurred on a weekend.")
+            return None
+        elif weekdays[zoom_time.weekday()] not in schedule['Days']:
+            logger.debug("No opencast recording scheduled for this day of the week.")
+            return None
+
+        scheduled_time = datetime.strptime(schedule['Time'], '%H:%M')
+        scheduled_time = zoom_time.replace(hour=scheduled_time.hour, minute=scheduled_time.minute)
+        timedelta = abs(zoom_time - scheduled_time).total_seconds()
+
+        minutes = 20
+        if timedelta < (minutes * 60):
```

Comment:
does the schedule indicate how long the class is? what if something happens and the recording starts 30m late? would it be possible to do something like:

    if class_start_time - 20m < zoom_time < class_end_time:
        ...

Reply:
Yes it does. Good idea.